src/CollapseEmbeds/hider.tsx

```diff
@@ -59,10 +61,10 @@ export const Hider = ({placeholders, type, children, id}: HiderProps): JSX.Eleme
     }, [shown, id]);
 
     Settings.useListener(({hideByDefault, collapsedStates}) => {
-        if (!id) {
-            setShown(!hideByDefault);
-        } else if (collapsedStates[id]) {
+        if (id && collapsedStates[id]) {
             setShown(!collapsedStates[id].collapsed);
+        } else {
+            setShown(!hideByDefault);
```

Comment:
Same thing as above. Check for the presence of an entry for `id`, not whether the entry is truthy.